projects/FW103/src/main.c

```diff
@@ -0,0 +1,52 @@
+#include <stdint.h>
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+GpioAddress reading_addr = { .port = GPIO_PORT_A, .pin = 6 };
+GpioAddress button_addr = { .port = GPIO_PORT_B, .pin = 2 };
+
+GpioSettings reading_settings = { .direction = GPIO_DIR_IN,
+                                  .state = GPIO_STATE_LOW,
+                                  .alt_function = GPIO_ALTFN_NONE,
+                                  .resistor = GPIO_RES_NONE };
+
+GpioSettings button_settings = { .direction = GPIO_DIR_IN,
+                                 .state = GPIO_STATE_LOW,
+                                 .alt_function = GPIO_ALTFN_NONE,
+                                 .resistor = GPIO_RES_NONE };
+
+InterruptSettings interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,
+  .priority = INTERRUPT_PRIORITY_NORMAL,
+};
+
+void interrupt_handler(const GpioAddress *address, void *context) {
+  AdcChannel *reading_channel = (AdcChannel *)context;
+  uint16_t data = 0;
+  adc_read_raw(*reading_channel, &data);
```

Comment:
The homework does say:
> Create a project that reads an ADC **converted** reading...

So use `adc_read_converted` instead. The difference is that `adc_read_converted` gives you output in mV, whereas `adc_read_raw` gives you whatever raw junk came out of the ADC. You almost always want `adc_read_converted`.